iphone/CoreApi/CoreApi/Logger/Logger.mm

```diff
@@ -132,10 +115,54 @@ bool AssertMessage(base::SrcPoint const & src, std::string const & message)
 
 // MARK: - Private
 
-+ (BOOL)isFileLoggingEnabled {
-  return [self logger].fileHandle != nil;
++ (void)enableFileLogging {
+  Logger * logger = [self logger];
+  NSFileManager * fileManager = [NSFileManager defaultManager];
+
+  // Create a log file if it doesn't exist and setup file handle for writing.
+  NSString * filePath = [[fileManager temporaryDirectory] URLByAppendingPathComponent:kLogFileName].path;
+  if (![fileManager fileExistsAtPath:filePath]) {
+    [fileManager createFileAtPath:filePath contents:nil attributes:nil];
+  }
+  NSFileHandle * fileHandle = [NSFileHandle fileHandleForWritingAtPath:filePath];
+  if (fileHandle == nil) {
+    LOG(LERROR, ("Failed to open log file for writing"));
+    [self setFileLoggingEnabled:NO];
+    return;
+  }
+  // Clean up the file if it exceeds the maximum size.
+  if ([fileManager contentsAtPath:filePath].length > kMaxLogFileSize)
+    [fileHandle truncateFileAtOffset:0];
+
+  logger.filePath = filePath;
+  logger.fileHandle = fileHandle;
+  LOG(LINFO, ("File logging is enabled"));
 }
- 
+
++ (void)disableFileLogging {
+  Logger * logger = [self logger];
+
+  [logger.fileHandle closeFile];
+  logger.fileHandle = nil;
+  logger.fileSize = 0;
+  NSError * error;
+  [NSFileManager.defaultManager removeItemAtPath:logger.filePath error:&error];
+  if (error)
+    LOG(LERROR, ("Failed to remove log file"));
+  LOG(LINFO, ("File logging is disabled"));
+}
+
+- (BOOL)isFileLoggingEnabled {
+  return self.fileHandle != nil;
+}
+
+- (void)notifyOnFileDidChangeSize:(uint64_t)fileSize {
```

Comment:
Why are this method and observer needed? Can we drop them?

Reply:
This is needed to update the log file size label in the Settings screen.

Reply:
A real-time observing is not needed here. Can the file size be queried and set once?

Reply:
Sure!